### PR TITLE
add hw_10

### PR DESCRIPTION
**Pull request by baranovskyi-a**
Снимок экрана 2024-10-30 в 23 09 00
Снимок экрана 2024-10-30 в 23 08 17

### CODE REVIEW

File: lesson_10/process_sales_gcp.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        Params:
+            ds - date in "yyyy-mm-dd" format
+            raw_dir - dir to save
+    '''
+    page = 1
+    records = []
+    while True:
+        # request current page
+        response = requests.get(
+            url='https://fake-api-vycpfa6oca-uc.a.run.app/sales',
```
Comment (tmikhailovska) [inline lesson_10/process_sales_gcp.py]: It is better to store URL parameters as function parameters and pass

File: lesson_10/process_sales_gcp.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            params={'date': ds, 'page': page},
+            headers={'Authorization': AUTH_TOKEN},
+            timeout=REQUEST_TIMEOUT_SECONDS
+        )
+        # break if there are no more records for current date
+        if response.status_code == 404:
+            break
```
Comment (tmikhailovska) [inline lesson_10/process_sales_gcp.py:49]: should to add an Airflow exception if page don't exist, should know about it and stop the pipelines

**Review by tmikhailovska — APPROVED** [review]

File: lesson_10/process_sales_gcp.py (hunk 2 above)
Comment (baranovskyi-a) [reply lesson_10/process_sales_gcp.py:49]: however, encountering a 404 status code is expected behavior here and code shouldn't be interrupted

File: lesson_10/process_sales_gcp.py (hunk 1 above)
Comment (baranovskyi-a) [reply lesson_10/process_sales_gcp.py]: done

File: lesson_10/process_sales_gcp.py (hunk 2 above)
Comment (tmikhailovska) [reply lesson_10/process_sales_gcp.py:49]: agree

**Pull request merged**